### ISSUE TITLE
indexPath in didSelectRowAtIndexPath

### ISSUE THREAD

**Issue by nyeu** [Krelborn/KILabel#30]
I have a tableview where cells display a KILabel, when I tapped a cell (not the kilabel link), I get the wrong indexpath, I get the previous indexPath selected. 
If I remove the kilabel, everything works like a charm.

**Comment by nyeu** [comment]
The way to "solve" the issue is adding [tableView reloadData] at the end of didSelectRowAtIndexPath.

**Comment by Krelborn** [comment]
That doesn't sound right. You really don't want to be reloading the table data after making a selection.

Have you looked at the demo project to see how to use it in a table. I can't imagine why using KILabel would affect selection of table cells.

**Comment by nyeu** [comment]
Yes, I know it doesn't sound right but it was the only way to get KILabel working. I looked at the demo project and all the examples available and all looks right. The thing is, if it was my mistake, why with the reloadData worked?
It's a weird thing.

**Comment by vgolovnev** [comment]
I have exactly the same problem. It seems like a bug.

**Comment by dayj** [comment]
I ran into this same issue. It has to do with the way touchesBegan/Moved/Ended/Cancelled have been implemented. I was able to resolve the issue by using a UITapGestureRecognizer in lieu of overriding the touch event methods.

**Comment by kanzitdinov** [comment]
I had the same problem. I added my KILabel with addSubview on the cell. And only one easy and painless solution I found is to add next code to end of method didSelectRowAtIndexPath
```
[tableView reloadRowsAtIndexPaths:@[ indexPath ] withRowAnimation:UITableViewRowAnimationNone]; 
```

**Comment by fealonelei** [comment]
@nyeu @vgolovnev @Dayj  
in *\* - (void)touchesEnded:(NSSet _)touches withEvent:(UIEvent *)event  *_ method , line 705, perhaps, 
just comment out  **[super touchesBegan:touches withEvent:event];** or remove it. 
It will fix the problem and didSelectRowAtIndexPath works happy again.

**Comment by LinkRober** [comment]
Thanks @dayj
